Solution for Assignment 1

jyotsnaven opened the pull request:
@jpang-h4, @duaneatat @mikwat @ali-tetration

 Please review my first assignment

duaneatat commented on assignment1/assignment1-solution/app.js:
```diff
@@ -0,0 +1,54 @@
+};
+
+function getLunchStatus(inputString, separator) {
+  var arrayOfDishes = inputString.split(separator);
```
any reason to prefer the dishes as a delimited string rather than an array to begin with? Also, will these strings be trimmed of surrounding spaces?

duaneatat commented on assignment1/assignment1-solution/app.js:
```diff
@@ -0,0 +1,54 @@
+    if(numberOfDish < 4){
+      return "Enjoy";
+    }
+    else if(numberOfDish > 3){
```
probably don't need the conditional, `else` is sufficient

duaneatat commented on assignment1/assignment1-solution/index.html:
```diff
@@ -0,0 +1,42 @@
+
+         </div>
+         <div class="form-group">
+             <button class="btn btn-default" ng-click='status.displayStatus()'>Check If Too Much</button>
```
in order to allow the form to be submitted if the user hits 'enter', you can wrap the form in a `form` element and use the `ng-submit` directive (rather than `ng-click` here) and use `type="submit"`

duaneatat commented on assignment1/assignment1-solution/index.html:
```diff
@@ -0,0 +1,42 @@
+         <div class="form-group">
+             <button class="btn btn-default" ng-click='status.displayStatus()'>Check If Too Much</button>
+         </div>
+         <div class="form-group message" id = "pass">
```
no spaces in `id = "pass"`, also ids should be used sparingly, and it should be guaranteed that there is only one per-page. Fine in this example but in a more complex application these ids might be too general, just something to keep in mind

jyotsnaven commented:
@duaneatat @mikwat @ali-tetration @jpang-h4 

Made the following changes:
Integrated JSHint with IDE and validated code
Two-space indentation for code inside function() and other inner blocks
Changed strings from “ “ to ‘ ‘
Moved use strict two lines above function() and made it a function
Replaced old code with html form, ng-submit and input-type=‘submit’.
Removed redundant condition
Replaced CSS ids with <style> tags for better design

jpang-h4 commented on assignment1/assignment1-solution/index.html:
```diff
@@ -16,26 +16,25 @@
+           <div class="form-group">
+               <input type="submit" class="btn btn-default" value="Check if too much!"></input>
+           </div>
+           <div class="form-group message" style="color:green">
```
The convention is to try not to use `style` tag. 

Try to create classes in style.css called `message-success` and `message-error` to specify colors?

duaneatat replied:
+1

jpang-h4 commented on assignment1/assignment1-solution/app.js:
```diff
@@ -1,54 +1,55 @@
-(function () {
-'use strict';
+'use strict()';
```
I think this is just a string not a function `'use strict';`

jpang-h4 commented on assignment1/assignment1-solution/app.js:
```diff
@@ -0,0 +1,55 @@
+        return 'Too much!';
+      }
+    }
+    //To handle null input
```
Strictly speaking, this else block gets hit with either 'null' or 'undefined' value. 

be able to distinguish null vs undefined. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/null

Also see strict inequality operators https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Comparison_Operators

![image](https://cloud.githubusercontent.com/assets/7469575/22614717/f8589d04-ea3a-11e6-822b-7052536707cf.png)

duaneatat commented on assignment1/assignment1-solution/app.js:
```diff
@@ -0,0 +1,55 @@
+    if (inputString){
+      /*Implement this case `item 1, item2,,item3`
+      or this case `item 1, item2, ,item3` */
+      for(var dish in arrayOfDishes){
```
nit space after `for` and after paren: `for (var dish in arrayOfDishes) {`

duaneatat commented on assignment1/assignment1-solution/app.js:
```diff
@@ -0,0 +1,55 @@
+          numberOfDish++;
+        }
+      }
+      if(numberOfDish < 4){
```
nit `if (numberOfDish < 4) {` not sure if our jshint catches these

duaneatat commented on assignment1/assignment1-solution/index.html:
```diff
@@ -0,0 +1,41 @@
+           <div class="form-group message" style="color:green">
+              {{status.validMessage}}
+           </div>
+           <div class="form-group message" style="color:red">
```
same here, no inline-style

duaneatat commented on assignment1/assignment1-solution/index.html:
```diff
@@ -0,0 +1,41 @@
+     <h1>Lunch Checker</h1>
+      <!-- controllerAs Pattern -->
+     <div ng-controller="StatusController as status">
+         <form ng-submit="status.displayStatus()">
```
nit check indentation

duaneatat commented on assignment1/assignment1-solution/styles/style.css:
```diff
@@ -0,0 +1,4 @@
+.message{
+  font-size: 1.3em;
```
👍
